cmd/server/controllers/EmployeeController.go

```diff
@@ -0,0 +1,86 @@
+package controllers
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/vinigracindo/mercado-fresco-stranger-strings/internal/domains/employees"
+)
+
+type EmployeeController struct {
+	service employees.Service
+}
+
+func NewEmployee(service employees.Service) EmployeeController {
+	return EmployeeController{service: service}
+}
+
+func (controller EmployeeController) GetAll() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		employees, err := controller.service.GetAll()
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{
+			"data": employees,
+		})
+	}
+}
+
+func (controller EmployeeController) Get() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		id, err := strconv.ParseInt(c.Param("id"), 10, 64)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		employee, err := controller.service.Get(id)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{
+			"data": employee,
+		})
+	}
+}
+
+func (controller EmployeeController) Store() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var req request
+		if err := c.ShouldBindJSON(&req); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
+				"error":   err.Error(),
+				"message": "Invalid request",
+			})
+			return
+		}
+
+		employee, err := controller.service.Store(req.CardNumberId, req.FirstName, req.LastName, req.WarehouseId)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
```

Comment:
Mesma dúvida: aqui vem o erro que foi tratado lá no repository, certo?
Se for, podemos retornar um Conflict ao invés de StatusInternalServerError?